File: app/src/main/java/com/zhiyong/tingxie/CommonViewModel.java
package com.zhiyong.tingxie;

import android.app.Application;
import androidx.lifecycle.AndroidViewModel;

import com.zhiyong.tingxie.db.QuizPinyin;

// todo: Try placing duplicate functions of QuizViewMode and WordViewModel.
public abstract class CommonViewModel extends AndroidViewModel {
    protected QuizRepository mRepository;

    public CommonViewModel(Application application, int quizId) {
        super(application);
        mRepository = new QuizRepository(application, quizId);
    }

    void addQuizPinyin(QuizPinyin quizPinyin) {
        mRepository.insertQuizPinyin(quizPinyin);
    }

    QuizRepository getRepository() {
        return mRepository;
    }
}

File: app/src/main/java/com/zhiyong/tingxie/CommonViewModel.kt
package com.zhiyong.tingxie

import android.app.Application
import androidx.lifecycle.AndroidViewModel
import com.zhiyong.tingxie.db.QuizPinyin

// todo: Try placing duplicate functions of QuizViewMode and WordViewModel.
abstract class CommonViewModel(application: Application, quizId: Int) : AndroidViewModel(application) {
    var repository: QuizRepository
        protected set

//    fun addQuizPinyin(quizPinyin: QuizPinyin?) {
//        repository.insertQuizPinyin(quizPinyin)
//    }

    init {
        repository = QuizRepository(quizId.toLong(), application)
    }
}